source/crud/read-operations/text.txt

```diff
@@ -0,0 +1,260 @@
+===========
+Search Text
+===========
+
+.. default-domain:: mongodb
+
+Use the `$text
+<https://docs.mongodb.com/manual/reference/operator/query/text/>`_
+operator to perform text searches on fields which have a
+`text index <https://docs.mongodb.com/manual/core/index-text/>`_.
+
+Overview
+--------
+
+Text searches let you search a collection of strings for words or
+phrases, instead of matching substrings like
+`regular expresssions
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions>`_,
+also known as RegEx. While regex is very powerful, and MongoDB even
+offers a `$regex <>` operator, text searches work particularly well when
+applied to unstructured text, like transcripts, essays, or even web
+pages. Because MongoDB's text indexes are aware of concepts like
+plurality, case sensitivity, stop words, whitespace, and punctuation in
+multiple languages, all you have to do is query with a search term to
+find documents that contain your term.
```

Comment:
Pretty solid!

- What's a "collection of strings"? Is this really "string-valued fields"?
- Maybe we could have a blurb about why `$text` is more performant than `$regex`?
- nit: I'd avoid starting a sentence with "Because" - you can probably invert the sentence from "Because REASON, FOO" to "FOO because REASON".
- nit: "all you have to do" is like "simple" and "easy" - probably better to avoid that type of phrasing altogether

Reply:
- switched to "string-valued fields"
- I would hesitate to imply that $text is always more performant than regex -- regex is darn fast. Regex on an indexed field is probably in the same league as $text, though obviously it has different use cases. $text offers enough usability advances for certain purposes that I don't think we need to worry about performance differences.
- I tried inverting this sentence and I actually really prefer starting it with "because", though obviously that's subjective. If you feel strongly I can always change it despite that.
- I understand our aversion to "simple" and "easy" (because it can seem condescending to an end user on either a conscious or unconscious level) but I can't think of a better way to structure this thought. Would appreciate any suggestions on this.